src/main.js

```diff
@@ -55,23 +61,33 @@ if (crashReporterDirectory) {
 		}
 	}
 
-	// Crashes are stored in the temp directory by default, so we
+	// Crashes are stored in the crashDumps directory by default, so we
 	// need to change that directory to the provided one
-	console.log(`Found --crash-reporter-directory argument. Setting temp directory to be '${crashReporterDirectory}'`);
-	app.setPath('temp', crashReporterDirectory);
-
-	// Start crash reporter
-	const { crashReporter } = require('electron');
-	const productName = (product.crashReporter && product.crashReporter.productName) || product.nameShort;
-	const companyName = (product.crashReporter && product.crashReporter.companyName) || 'Microsoft';
-	crashReporter.start({
-		companyName: companyName,
-		productName: process.env['VSCODE_DEV'] ? `${productName} Dev` : productName,
-		submitURL: '',
-		uploadToServer: false
-	});
+	console.log(`Found --crash-reporter-directory argument. Setting crashDumps directory to be '${crashReporterDirectory}'`);
+	app.setPath('crashDumps', crashReporterDirectory);
+} else {
+	const appCenter = product.appCenter;
+	// Disable Appcenter crash reporting if
+	// * --crash-reporter-directory is specified
+	// * disable-crash-reporter runtime argument is set to 'true'
+	// * --disable-crash-reporter command line parameter is set
+	if (appCenter && (argvConfig['disable-crash-reporter'] === false || argvConfig['disable-crash-reporter'] === 'false') &&
+		!app.commandLine.hasSwitch('disable-crash-reporter')) {
```

Comment:
Where is this coming from, Chrome?

Reply:
No not from chrome but If users start the app with `--disable-crash-reporter`. Isn't that still a supported way to disable it, useful for ci/testing purposes too.

Reply:
@deepak1556 yeah I got confused why you go through `app.commandLine` if all the command args are accessible via the `args` variable already above?

Reply:
oh yeah should use that, habitual api call :)